bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandRestClient.java

```diff
@@ -0,0 +1,196 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TouchWandRestClient} is responsible for handling low level commands units TouchWand WonderFull hub
+ * REST API interface
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@NonNullByDefault
+public class TouchWandRestClient {
+
+    private final Logger logger = LoggerFactory.getLogger(TouchWandRestClient.class);
+
+    static CookieManager cookieManager = new CookieManager();
+
+    private String touchWandIpAddr = "";
+    private String touchWandPort = "";
+    private boolean isConnected = false;
+
+    private static final HttpMethod METHOD_GET = HttpMethod.GET;
+    private static final HttpMethod METHOD_POST = HttpMethod.POST;
+
+    private static final String CMD_LOGIN = "login";
+    private static final String CMD_LIST_UNITS = "listunits";
+    private static final String CMD_LIST_SCENARIOS = "listsencarios";
+    private static final String CMD_UNIT_ACTION = "action";
+    private static final String CMD_GET_UNIT_BY_ID = "getunitbyid";
+
+    private static final String ACTION_SWITCH_OFF = "{\"id\":%s,\"value\":" + SWITCH_STATUS_OFF + "}";
+    private static final String ACTION_SWITCH_ON = "{\"id\":%s,\"value\":" + SWITCH_STATUS_ON + "}";
+    private static final String ACTION_SHUTTER_DOWN = "{\"id\":%s,\"value\":0,\"type\":\"height\"}";
+    private static final String ACTION_SHUTTER_UP = "{\"id\":%s,\"value\":255,\"type\":\"height\"}";
+    private static final String ACTION_SHUTTER_STOP = "{\"id\":%s,\"value\":0,\"type\":\"stop\"}";
+    private static final String ACTION_SHUTTER_POSITION = "{\"id\":%s,\"value\":%s}";
+    private static final String ACTION_DIMMER_POSITION = "{\"id\":%s,\"value\":%s}";
+
+    private static final String CONTENT_TYPE_APPLICATION_JSON = MimeTypes.Type.APPLICATION_JSON.asString();
+
+    private static final int REQUEST_TIMEOUT_SEC = 10000; // 10 seconds
+    private Map<String, String> commandmap = new HashMap<String, String>();
+
+    private HttpClient httpClient;
+
+    public TouchWandRestClient(HttpClient httpClient) {
+        commandmap.put(CMD_LOGIN, "/auth/login?");
+        commandmap.put(CMD_LIST_UNITS, "/units/listUnits");
+        commandmap.put(CMD_LIST_SCENARIOS, "/scenarios/listScenarios");
+        commandmap.put(CMD_UNIT_ACTION, "/units/action");
+        commandmap.put(CMD_GET_UNIT_BY_ID, "/units/getUnitByID?");
+
+        this.httpClient = httpClient;
+    }
+
+    public final boolean connect(String user, String pass, String ipAddr, String port) {
+        touchWandIpAddr = ipAddr;
+        touchWandPort = port;
+        isConnected = cmdLogin(user, pass, ipAddr);
+
+        return isConnected;
+    }
+
+    @SuppressWarnings("null")
+    private final boolean cmdLogin(String user, String pass, String ipAddr) {
+        String command = buildUrl(CMD_LOGIN) + "user=" + user + "&" + "psw=" + pass;
+        String response = sendCommand(command, METHOD_GET, "");
```

Comment:
You should annotate `sendCommand()` with `@NonNull` if it can return a null value. Then, the null suppression above could be removed.